### PR TITLE
Use mod.conf for dependencies and description

### PR DESCRIPTION
**Pull request by Panquesito7**
Deletes deprecated `description.txt`.
This will only work with MT/MTG 5.0.0 and above.

### CODE REVIEW

**Comment by raymoo** [comment]
Any reason for removing newlines at the end of every file? Ending every line with a newline is the POSIX standard for text files.

File: standard_monoids.lua [hunk 1]
```diff
@@ -63,7 +63,6 @@ player_monoids.jump = monoid({
 	end,
 })
 
-
```
Comment (raymoo) [inline standard_monoids.lua:66]: I agree that these newlines are unnecessary, but could you put style fixes in a separate commit? It's ok if it's in the same PR as the `mod.conf` changes.

File: init.lua [hunk 2]
```diff
@@ -115,4 +115,4 @@ function mon_meta:value(player)
 end
 
 dofile(modpath .. "standard_monoids.lua")
-dofile(modpath .. "test.lua")
+dofile(modpath .. "test.lua")
```
Comment (raymoo) [inline init.lua:118]: I don't think you should remove newlines from the last line of every file. Newlines at the end of files is part of the POSIX standard for text files, so removing them would make the files technically not portable (though all editors will support it anyway). More importantly, the newlines will get readded when someone changes the files from a POSIX-compliant system.

File: test.lua [hunk 3]
```diff
@@ -4,6 +4,7 @@ local speed = player_monoids.speed
 minetest.register_privilege("monoid_master", {
 	description = "Allows testing of player monoids.",
 	give_to_singleplayer = false,
+	give_to_admin = true,
```
Comment (raymoo) [inline test.lua:7]: This is fine but should be a separate commit. It's fine if you include it in this PR.

**Review by raymoo — CHANGES_REQUESTED** [review]
Partially redundant with my previous comment but after looking through the changes I thought I should collect all my comments in a review.

**Comment by Panquesito7** [comment]
I don't wan't to be rude, but, what's wrong with having all changes in a single commit?

> Any reason for removing newlines at the end of every file?

IMHO, they're useless; I can un-do those changes if you wan't.

**Comment by raymoo** [comment]
>I don't wan't to be rude, but, what's wrong with having all changes in a single commit?

Each logical change should be its own commit so that they can be dealt with independently by operations such as cherry picking or reversions. This also includes bisection, where it will be harder to determine the cause of a bug if you land on a commit with multiple changes in it.

It also makes it easier to browse commits. Not having multiple changes means needing to cram less information into the commit message, which (IMO) makes it easier to read when viewing the commit log. Having just the file changes for one thing also makes it faster to find the files relevant to whatever feature / bug fix / other logical change you want to examine.

>IMHO, they're useless; I can un-do those changes if you wan't.

Like I mentioned earlier, newlines at the end of each line is part of the POSIX standard and most editors on Linux or Mac will automatically add them to make valid text files. Removing will result in thrashing where the next change made by someone on a POSIX system will re-add the newlines.

**Comment by Panquesito7** [comment]
@raymoo, I'll create a new PR and fix all those issues.

**Pull request closed without merging**